### ISSUE TITLE
MSL: Feature request: Add a "enable_decoration_location" option.

### ISSUE THREAD

**Issue by chirsz-ever** [KhronosGroup/SPIRV-Cross#1816]
I want to convert some SPIR-V modules from OpenGL GLSL and `GL_ARB_gl_spirv`, in which there are default uniforms with `DecorationLocation`, and I want keep them as MSL's buffer index.

Currently the `CompilerMSL` does not provide a function to do this kind of auto-remap, and implementing it with `add_msl_resource_binding` is just not a very easy work. And, `CompilerMSL::get_metal_resource_index` is not a virtual function.

So is it possible to add a `enable_decoration_location` option in `CompilerMSL::Options` just like `enable_decoration_binding`? If  it is approved, I could make a PR to accomplish it.

**Comment by chirsz-ever** [comment]
Or, could `enable_decoration_binding` choose the `DecorationLocation` if there is no `DecorationBinding`?

**Comment by HansKristian-Work** [comment]
By default uniforms you mean the legacy GL uniforms ala:
```
layout(location = 4) uniform float foo;
```
?
FWIW, that's not supported in Vulkan at all, and I'd be surprised if spirv-cross even supports that.

**Comment by chirsz-ever** [comment]
Yes, SPIRV-Cross just support, and glslang also support to convert GLSL source code to this kind of SPIR-V code. This is especially convenient for converting GLSL to MSL, while the Vulkan-style SPIR-V code which uses std140 layout would cause some problems.
